UriGenerator.php

```diff
@@ -106,4 +123,16 @@ public function resolveConflict(UriContext $uriContext)
 
         return $uri;
     }
+
+    /**
+     * Configure options which apply to each token provider
+     *
+     * @param OptionsResolver
+     */
+    private function configureGlobalOptions(OptionsResolver $optionsResolver)
+    {
+        $optionsResolver->setDefaults(array(
+            'collapse_empty' => false,
```

Comment:
Not sure I undetstand your last point. But thinking about I can't see a situation where you would want to allow empty and not collapse slashes, so +1 for `allow_empty`.